### ISSUE TITLE
PerformanceManager in simulator

### ISSUE THREAD

**Issue by prydin** [vmware/govmomi#1207]
Are there any plans on implementing the PerformanceManager methods in the simulator? Would you be interested in a contribution on that?

**Comment by prydin** [comment]
OK. I will work on this. I'm assuming it's best to just fork from master?

**Comment by dougm** [comment]
Excellent.  Yes, fork from master, we don't have any other branches currently.

**Comment by prydin** [comment]
I'm making progress. As you can see, all metrics come back as "1", but I'm planning on providing some more meaningful metrics. Probably a time series from some lab, repeated over and over again...

```
prydin-a02:vcharness prydin$ govc metric.ls vm/* | grep cpu
cpu.usage.average
cpu.used.summation
cpu.system.summation
cpu.overlap.summation
cpu.latency.average
cpu.idle.summation
cpu.entitlement.latest
cpu.run.summation
cpu.swapwait.summation
cpu.wait.summation
cpu.usagemhz.average
cpu.costop.summation
cpu.demandEntitlementRatio.latest
cpu.readiness.average
cpu.demand.average
cpu.maxlimited.summation
cpu.ready.summation
rescpu.maxLimited1.latest
rescpu.actpk15.latest
rescpu.runpk15.latest
rescpu.actav1.latest
rescpu.runav5.latest
rescpu.runav1.latest
rescpu.actav15.latest
rescpu.maxLimited5.latest
rescpu.runav15.latest
rescpu.sampleCount.latest
rescpu.actpk5.latest
rescpu.actpk1.latest
rescpu.samplePeriod.latest
rescpu.actav5.latest
rescpu.runpk5.latest
rescpu.runpk1.latest
rescpu.maxLimited15.latest
prydin-a02:vcharness prydin$ govc metric.sample /DC0/vm/DC0_C0_RP0_VM0 cpu.usage.average
DC0_C0_RP0_VM0  *  cpu.usage.average    0.01,0.01,0.01,0.01,0.01,0.01  %
prydin-a02:vcharness prydin$ 
```

**Comment by prydin** [comment]
Here's the fork if you want to peek. https://github.com/prydin/govmomi

**Comment by prydin** [comment]
I have successfully implemented the following methods: QueryAvailablePerfMetric, QueryPerf, QueryPerfCounter and QueryPerfProviderSummary. I'm not planning on implementing the interval manipulation methods or QueryPerfComposite. Is anyone really using the latter?

You think this is enough for "minimal viable product" commit or do you think we need more methods implemented?

**Comment by dougm** [comment]
Fixed by #1230 , thanks again @prydin
